 <launch>

<arg name="robot_reference_frame"          default="base_footprint"/> 
<arg name="child_HD_frame_name"          default="omni_rotation"/> 
<!-- Change these when changing different robot reference frame!!!! -->
<arg name="rot_matrix_array"          default="0 0 -1 -1 0 0 0 1 0"/> <!-- base_footprint -->
<!--<arg name="rot_matrix_array"          default="0 -1 0 0 0 1 -1 0 0"/>--> <!-- wrist_ft_tool_link -->


<arg name="robot_base_frame"          default="base_footprint"/> 
<arg name="end_effector_frame"          default="wrist_ft_tool_link"/> 

<arg name="virtual_marker_name"          default="virtual_marker_transform"/>

<arg name="lock_state_topic" default="/omni1_lock_state"/> 

<!--<arg name="robot_reference_frame"          value="base_footprint"/> -->


 <include file="$(find omni_marco_launcher)/launch/omni_fake_marco.launch"> 
    <arg name="robot_base_frame"     default="$(arg robot_base_frame)"/>
    <arg name="end_effector_frame"     default="$(arg end_effector_frame)"/> 
 </include>

  <include file="$(find haptic_device_rotation)/launch/haptic_device_rotation.launch">  
   <arg name="parent_reference_frame"     default="$(arg robot_reference_frame)"/>  
   <arg name="child_HD_frame_name"     default="$(arg child_HD_frame_name)"/>
   <arg name="rot_matrix_array"     default="$(arg rot_matrix_array)"/>
 </include>

 <include file="$(find haptic_device_2_marker)/launch/omni_2_marker.launch">  
    <arg name="robot_reference_frame"     default="$(arg robot_reference_frame)"/>
    <arg name="HD_frame_name"     default="$(arg child_HD_frame_name)"/>
    <arg name="end_effector_frame"     default="$(arg end_effector_frame)"/>
 </include>

 <include file="$(find stiffness_commanding)/launch/stiffness_commanding.launch"> 
    <arg name="end_effector_frame"     default="$(arg end_effector_frame)"/> 
    <arg name="virtual_marker_name"     default="$(arg virtual_marker_name)"/>
    <arg name="lock_state_topic" value="$(arg lock_state_topic)"/>
 </include>
 </launch>
